code/scripting/api/libs/base.cpp

```diff
@@ -106,6 +106,82 @@ ADE_FUNC(createVector, l_Base, "[number x, number y, number z]", "Creates a vect
 	return ade_set_args(L, "o", l_Vector.Set(v3));
 }
 
+ADE_FUNC(createSurfaceNormal,
+	l_Base,
+	"[vector point1, vector point2, vector point3]",
```

Comment:
The optional specifier here does not match the one passed to `ade_get_args`. Here every parameter is optional but below you specify that everything is required.

Reply:
Oh, oops.